src/Drawer/Drawer.js

```diff
@@ -92,34 +87,244 @@ class Drawer extends React.Component {
+      const translateRatio = this.isHorizontalSwiping()
+        ? this.getTranslate(currentX) / this.getMaxTranslate()
+        : this.getTranslate(currentY) / this.getMaxTranslate();
+
+      this.maybeSwiping = false;
+      const swiping = this.state.swiping;
+      this.setState({
+        swiping: null,
+      });
+
+      // We have to open or close after setting swiping to null,
+      // because only then CSS transition is enabled.
+      if (translateRatio > 0.5) {
+        if (swiping === 'opening') {
+          this.setPosition(this.getMaxTranslate());
+        } else if (this.props.onClose != null) {
+          this.props.onClose();
+        }
+      } else if (swiping === 'opening') {
+        if (this.props.onOpen != null) {
+          this.props.onOpen();
+        }
+      } else {
+        this.setPosition(0);
+      }
+    } else if (this.maybeSwiping) {
+      if (!this.props.open && event != null) {
+        event.preventDefault(); // prevent ghost clicks
+      }
+      this.maybeSwiping = false;
+      this.forceUpdate();
+    }
+
+    this.removeBodyTouchListeners();
+  };
+
+  getAnchor() {
+    return this.props.theme.direction === 'rtl' && this.isHorizontalSwiping()
+      ? oppositeDirection[this.props.anchor]
+      : this.props.anchor;
+  }
+
+  getMaxTranslate() {
+    return this.isHorizontalSwiping() ? this.drawer.clientWidth : this.drawer.clientHeight;
+  }
+
+  getTranslate(current) {
+    const swipeStart = this.isHorizontalSwiping() ? this.swipeStartX : this.swipeStartY;
+    return Math.min(
+      Math.max(
+        this.state.swiping === 'closing'
+          ? -(current - swipeStart)
+          : this.getMaxTranslate() + (swipeStart - current) - this.props.swipeAreaWidth,
+        0,
+      ),
+      this.getMaxTranslate(),
+    );
+  }
+
+  setPosition(translate) {
+    const rtlTranslateMultiplier = ['right', 'bottom'].indexOf(this.props.anchor) !== -1 ? 1 : -1;
+    const transformCSS = this.isHorizontalSwiping()
+      ? `translate(${rtlTranslateMultiplier * translate}px, 0)`
+      : `translate(0, ${rtlTranslateMultiplier * translate}px)`;
+    this.drawer.style.transform = transformCSS;
+    this.drawer.style.webkitTransform = transformCSS;
+
+    this.backdrop.style.opacity = 1 - translate / this.getMaxTranslate();
+  }
+
+  isHorizontalSwiping() {
+    return ['left', 'right'].indexOf(this.props.anchor) !== -1;
+  }
+
+  enableSwipeHandling() {
+    document.body.addEventListener('touchstart', this.onBodyTouchStart);
+  }
+
+  disableSwipeHandling() {
+    document.body.removeEventListener('touchstart', this.onBodyTouchStart);
+  }
+
+  removeBodyTouchListeners() {
+    document.body.removeEventListener('touchmove', this.onBodyTouchMove);
+    document.body.removeEventListener('touchend', this.onBodyTouchEnd);
+    document.body.removeEventListener('touchcancel', this.onBodyTouchEnd);
   }
 
   render() {
     const {
-      anchor: anchorProp,
+      anchor: anchorProp, // eslint-disable-line
```

Comment:
Can you remove the `// eslint-disable-line` comments? We shouldn't need them. It's the first time we use it on the codebase.